src/sign/mod.rs

```diff
@@ -293,6 +376,17 @@ impl CoseSign1 {
             self.payload.as_ref().unwrap_or(&vec![]),
         )
     }
+
+    /// Construct the to-be-signed data for this object, using a detached payload.
+    fn tbs_detached_data(&self, payload: &[u8], aad: &[u8]) -> Vec<u8> {
```

Comment:
As above, need to check that the payload is *not* set.

Reply:
Added a documented panic when self.payload.is_some().